### ISSUE TITLE
CUDAExecutionProvider does not handle Clip on float16 tensor.

### ISSUE THREAD

**Issue by PeterL1n** [microsoft/onnxruntime#8070]
**Describe the bug**
CUDAExecutionProvider does not handle Clip on float16. It gets resorted to CPUExecutionProvider.

**Urgency**
None. But prefer it to be fixed.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- ONNX Runtime installed from (source or binary): pip
- ONNX Runtime version: 1.8.0
- Python version: 3.8.11
- Visual Studio version (if applicable): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: 11.2
- GPU model and memory: RTX 2060 Super

**To Reproduce**
Unable to provide a model right now. You can create a graph with a clip node with float16 input, min, and max value.

**Expected behavior**
Operation executed on CUDA

**Comment by PeterL1n** [comment]
Only has problem on opset=11, opset=12 works fine.

**Comment by faxu** [comment]
Are you able to export your model with opset 12? 

It looks like the support for clip CUDA kernel was added here: https://github.com/microsoft/onnxruntime/pull/2411 by @zhanghuanrong

**Comment by PeterL1n** [comment]
@faxu

Yeah opset 12 works for me and that's my workaround now. I am just reporting the bug here.
